qml/ConnectionMenu.qml

```diff
@@ -175,20 +175,23 @@ Item {
 
                 Text {
                     text: "UDP Host/Port:"
-                    color: Material.primary
+                    color: udpIp.isValid ? Material.primary : Material.color(Material.Red)
                 }
 
                 PingTextField {
                     id: udpIp
                     text: "192.168.2.2"
                     Layout.columnSpan:  2
                     Layout.fillWidth: true
+                    // This regex matches IPV4 Addresses and valid domain names.
+                    // See https://www.regextester.com/index.php?fam=110631
+                    // TODO: try to simplify/split this
+                    property bool isValid: text.match(/^(((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)|(([a-zA-Z0-9-]+\.){0,5}[a-zA-Z0-9-][a-zA-Z0-9-]+\.[a-zA-Z]{2,63}?)|[a-zA-Z0-9-]+)$/gm)
```

Comment:
Is this better than a if and 2~3 checks ?

Reply:
Just to make sure, is this part equal to:
```
var intput;
// Check for valid ipv4 IP address
regex = "^(?!\.)((^|\.)([1-9]?\d|1\d\d|2(5[0-5]|[0-4]\d))){4}$"
if(regex.check(input)) { output = intput; return; }

// check if it's a normal domain
if(input.split('.') <= 2) { output = intput; return; }
```

You could also break the `or` of this same regex in multiple regex to allow a better readability and maintainability.
```js
property bool isValid:
                        // Check for valid IPV4 (0.0.0.0, 192.168.0.2)
                        text.match(/^(?!\.)((^|\.)([1-9]?\d|1\d\d|2(5[0-5]|[0-4]\d))){4}$/gm)
                        // Check for valid host names (potato.com, smashed.potato.com.br, great-potato-of-doom.com)
                        || text.match(/^(([a-zA-Z0-9-]+\.){0,5}[a-zA-Z0-9-][a-zA-Z0-9-]+\.[a-zA-Z]{2,63}?)$/gm)
                        // Check for a valid host name (localhost, raspberrypi, potato)
                        || text.match(/^[a-zA-Z0-9-]*$/gm)
```